workflow/controller/operator.go

```diff
@@ -1676,6 +1685,29 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 	return node, nil
 }
 
+//Checks the template if it exceeds its deadline
+func (woc *wfOperationCtx) checkTemplateDeadline(tmpl *wfv1.Template, node *wfv1.NodeStatus) (*time.Time, error) {
+	woc.log.Debugf("Check Template deadline")
+	if node == nil {
+		return nil, nil
+	}
+
+	if tmpl.TimeoutDuration != nil {
+		tmplMaxDuration, err := parseStringToDuration(tmpl.TimeoutDuration.StrVal)
+		if err != nil {
+			return nil, err
+		}
+		deadline := node.StartedAt.Add(tmplMaxDuration)
+
+		if node.Phase == wfv1.NodePending && time.Now().After(deadline) {
+			return nil, fmt.Errorf("%s %s exceeded its deadline", node.Name, node.Type)
+		}
+		return &deadline, nil
+	}
+	return nil, nil
+
```

Comment:
```suggestion

	return nil, nil
```
This is easier to read